src/systemctl/systemctl-enable.c

```diff
@@ -263,6 +263,33 @@ int verb_enable(int argc, char *argv[], void *userdata) {
                            "  instance name specified.",
                            special_glyph(SPECIAL_GLYPH_BULLET));
 
+        if (streq(verb, "disable") && arg_scope == LOOKUP_SCOPE_USER) {
+                /* We emit the warning only if all the unit files are enabled in
+                 * global scope (same as we did for the install info warning). */
+
+                bool enabled_global_scope = true;
+
+                STRV_FOREACH(name, names) {
+                        UnitFileState state;
+
+                        r = unit_file_get_state(LOOKUP_SCOPE_GLOBAL, arg_root, *name, &state);
+                        if (r == -ENOENT)
+                                enabled_global_scope = false;
+                        else if (r < 0)
+                                return log_error_errno(r, "Failed to get unit file state for %s: %m", *name);
+
+                        if (!IN_SET(state, UNIT_FILE_ENABLED, UNIT_FILE_ENABLED_RUNTIME))
+                                enabled_global_scope = false;
+
+                        if (!enabled_global_scope)
+                                break;
+                }
+
+                if (enabled_global_scope && !arg_quiet && !arg_no_warn)
```

Comment:
When `arg_quiet` or `arg_no_warn` is set, the whole checking logic is not necessary. Please move the condition more earlier place.